src/controllers/authController.ts

```diff
@@ -0,0 +1,95 @@
+import jwt, { JwtPayload, TokenExpiredError } from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import { validationResult } from 'express-validator';
+import { Request, Response } from 'express';
+import { Types } from 'mongoose';
+
+import User from '../models/user.js';
+import Role from '../models/role.js';
+import { secret } from '../config/tokenKey.js';
+
+const generateAccessToken = (id: Types.ObjectId, roles: string[]) => {
```

Comment:
you need to move logic from controllers to `services`
services should contain only logic that processes something.
controllers in general should validate requests -> call methods from specific services -> return result